spec/rubocop/cop/style/file_read_spec.rb

```diff
@@ -0,0 +1,205 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Style::FileRead, :config do
+  it 'registers an offense for and corrects `File.open(filename).read`' do
+    expect_offense(<<~RUBY)
+      File.open(filename).read
+      ^^^^^^^^^^^^^^^^^^^^^^^^ Use `File.read`.
+    RUBY
+
+    expect_correction(<<~RUBY)
+      File.read(filename)
+    RUBY
+  end
+
+  it 'registers an offense for and corrects `::File.open(filename).read`' do
+    expect_offense(<<~RUBY)
+      ::File.open(filename).read
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^ Use `File.read`.
+    RUBY
+
+    expect_correction(<<~RUBY)
+      ::File.read(filename)
+    RUBY
+  end
+
+  it 'registers an offense for and corrects the `File.open` with symbolic read proc (implicit text mode)' do
+    expect_offense(<<~RUBY)
+      File.open(filename, &:read)
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use `File.read`.
+    RUBY
+
+    expect_correction(<<~RUBY)
+      File.read(filename)
+    RUBY
+  end
+
+  it 'registers an offense for and corrects the `::File.open` with symbolic read proc (implicit text mode)' do
+    expect_offense(<<~RUBY)
+      ::File.open(filename, &:read)
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use `File.read`.
+    RUBY
+
+    expect_correction(<<~RUBY)
+      ::File.read(filename)
+    RUBY
+  end
+
+  it 'registers an offense for and corrects the `File.open` with inline read block (implicit text mode)' do
+    expect_offense(<<~RUBY)
+      File.open(filename) { |f| f.read }
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use `File.read`.
+    RUBY
+
+    expect_correction(<<~RUBY)
+      File.read(filename)
+    RUBY
+  end
+
+  it 'registers an offense for and corrects the `::File.open` with inline read block (implicit text mode)' do
+    expect_offense(<<~RUBY)
+      ::File.open(filename) { |f| f.read }
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use `File.read`.
+    RUBY
+
+    expect_correction(<<~RUBY)
+      ::File.read(filename)
+    RUBY
+  end
+
+  it 'registers an offense for and corrects the `File.open` with multiline read block (implicit text mode)' do
+    expect_offense(<<~RUBY)
+      File.open(filename) do |f|
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^ Use `File.read`.
+        f.read
+      end
+    RUBY
+
+    expect_correction(<<~RUBY)
+      File.read(filename)
+    RUBY
+  end
+
+  it 'registers an offense for and corrects the `::File.open` with multiline read block (implicit text mode)' do
+    expect_offense(<<~RUBY)
+      ::File.open(filename) do |f|
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use `File.read`.
+        f.read
+      end
+    RUBY
+
+    expect_correction(<<~RUBY)
+      ::File.read(filename)
+    RUBY
+  end
```

Comment:
Subjectively, combining cbase check with every other example seems excessive.

Reply:
There's 4 cbase checks. Three are needed to cover the 3 branches of the node matcher. Shall I delete one?

Reply:
After adding @dvandersluis' suggestion, we now only have one branch, so I deleted all but one test.